src/main/java/hudson/plugins/active_directory/ActiveDirectoryAuthenticationProvider.java

```diff
@@ -68,6 +69,21 @@
      */
     private final _Connection con;
 
+    /**
+     * To specify the TTL and Size used for caching users and groups
+     */
+    public CacheConfiguration cache;
+
+    /**
+     * {@link UserDetails} cache.
+     */
+    private transient Map<String,CacheConfiguration.CacheEntry<UserDetails>> userCache = null;
```

Comment:
Why not create these maps on desireialization/construction?
it would avoid the need for synchronization blocks on `this` and then use something like a `ConcurrentHashMap`